thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2198,7 +2198,12 @@ class CHDRproportional: public CSimpleInterface, implements IHash
                 MemoryAttr ma;
                 activity->startInput(in);
                 if (activity->getOptBool(THOROPT_COMPRESS_SPILLS, true))
+                {
                     rwFlags |= rw_compress;
+                    StringBuffer compType;
+                    activity->getOpt(THOROPT_COMPRESS_SPILL_TYPE, compType);
+                    setCompFlag(compType, rwFlags);
+                }
                 Owned<IExtRowWriter> out = createRowWriter(tempfile, activity, rwFlags);
```

Comment:
Similar to previous comments re. CCompressedFile, createRowWriter takes a ICompressor, so it would be good if we can pass one here and for that to be the only compress info. needed.